File: 1635-number-of-good-pairs/1635-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int n=nums.length;
        HashMap<Integer,Integer> hm=new HashMap<>();
        int count=0;
        for(int i=0;i<n;i++){
            int val=nums[i];
            if(hm.containsKey(val)){
                count=count+hm.get(val);
                int curr=hm.get(val);
                hm.put(val,curr+1);
            }
            else{
                hm.put(val,1);
            }
            
        }
        return count;
        
    }
}

File: 1635-number-of-good-pairs/1635-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        dici={}
        for i in nums:
            if i not in dici:
                dici[i]=1
            else:
                dici[i]+=1
        ans=0
        for i in dici:
            n=dici[i]
            ans+=n*(n-1)//2
        return ans
        